### DISCUSSION TITLE
Retrieving value that can be string or null

### DISCUSSION THREAD

**Discussion by 7Pawns** [nlohmann/json#4410] [Q&A] [answered]
Might be a silly question but what is the proper way to do such thing?
I have been using `obj.value("field", "")` for a while thinking null will cause me to get the default value as well, but when the field value is null error `[json.exception.type_error.302] type must be string, but is object` pops up. Is this not the correct use for handling null in json fields?

**Comment by gregmarr** [comment, marked as answer]
No, the `value` function default only handles the case where the key isn't there, not when it's there but is a different type, and null is a type.  If you have a case where it can be null, then you need to handle that manually.  There isn't a built-in function that handles that case.
